.github/scripts/test-changed-packages.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -euo pipefail
+
+base=$(git merge-base origin/master HEAD)
+
+for pkg in $(git diff --name-only "$base" | grep '\.hcl$' | sed 's/\.hcl$//'); do
+  # Extract only NEW versions: versions in + lines that aren't in - lines
+  added=$(git diff "$base" -- "${pkg}.hcl" | grep '^+' | grep -oE '"[0-9]+\.[0-9]+\.[0-9]+"' | tr -d '"' | sort -u)
```

Comment:
I'm a bit concerned about false positives/negatives here. A more robust solution would be to checkout `$base`, dump all versions using `hermit search '^flyctl$' --json | jq -r '.[].Versions[]' | sort` to temporary files, then go back to `HEAD`, do the same, then diff? WDYT?

I'm not super fussed, so happy to go with this approach.

Reply:
Agreed, lemme try that approach.

Reply:
I ended up using worktrees vs a checkout. This ends up much more robust in exchange for +25 lines of bash. Thoughts @alecthomas?